### PR TITLE
DEVPROD-5757 Require owners, descriptions, and keywords in all workload files.

### PR DESCRIPTION
**Pull request by ahartschen**
**Jira Ticket:** DEVPROD-5757

**Whats Changed:**  
We now require that all workload files have these required fields. We made some exceptions for older files that did not have keywords. Any future workloads will require all the fields. I also changed things up to disable the formatting check because this was disabled about 2 years ago and there were ~13k errors that need corrected. I opened DEVPROD-5682 to have this addressed.

**Patch testing results:**  
Running the CLI locally, no output since there are no errors:

```
(genny_venv) ➜  genny git:(DEVPROD-5757) ./run-genny lint-yaml --no-format
(genny_venv) ➜  genny git:(DEVPROD-5757) 
```

**Workload Submission form:**  
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)

**Related PRs:**   
N/A

PLEASE READ AND REMOVE THE SECTION BELOW BEFORE SUBMITTING YOUR PR FOR REVIEW

1.  Once (1) you get approval from the STM or the Product Perf team for your change and (2) all your CI tests pass, you can initiate a merge by entering a comment on your PR with the text

    > evergreen merge

    This kicks your PR into the [evergreen commit queue][ecq]. Evergreen will automatically squash-merge your commit after checking that you have PR approval and that the CI tests all pass.

[ecq]: https://github.com/evergreen-ci/evergreen/wiki/Commit-Queue

### CODE REVIEW

**Review by thessem — APPROVED** [review]
From a perf infra point of view everything here looks good. I'd still wait for somebody from Product Perf on the test-config implications.

File: src/lamplib/src/genny/tasks/yaml_linter.py [hunk 1]
```diff
@@ -26,37 +32,47 @@ def main(genny_repo_root: str):
...
-def check_description(yaml_path):
+def check_required_fields(yaml_path):
     workload = _load_yaml(yaml_path)
-    if "Description" not in workload:
-        SLOG.error(f"Genny workload {yaml_path} lacks a Description field.")
-        return False
-    return True
+    no_errors = True
+    for field in REQUIRED_FIELDS:
+        if field not in workload:
```
Comment (MAhmadShah) [inline src/lamplib/src/genny/tasks/yaml_linter.py:70]: NIT: Do you think it's also worth checking that the field actually has some text in it? There could be a loop hole here where users can just put the key in with an empty value.

**Comment by MAhmadShah** [comment]
Just one comment other wise LGTM!

**Review by MAhmadShah — APPROVED** [review]

File: docs/using.md [hunk 2]
```diff
@@ -450,12 +450,14 @@ Try using `python test_result_summary.py --help` for more options.
 2.  Run the self-tests:
 
     ```bash
-	./run-genny lint-yaml  # Lint all YAML files
+	./run-genny lint-yaml  --no-format # Lint all YAML files
```
Comment (jimoleary) [inline docs/using.md:453]: Do we need ```--no-format``` here if the default is false?

File: src/lamplib/src/genny/tasks/yaml_linter.py [hunk 3]
```diff
@@ -26,37 +32,47 @@ def main(genny_repo_root: str):
...
-        yamllint_argv=yamllint_argv,
-    )
+        SLOG.info(
+            "Linting workload YAML files with yamllint",
+            count=len(all_yamls),
+            yamllint_argv=yamllint_argv,
+        )
 
-    yamllint.cli.run(yamllint_argv)
+        yamllint.cli.run(yamllint_argv)
```
Comment (jimoleary) [inline src/lamplib/src/genny/tasks/yaml_linter.py:60]: How are errors handled here? What do we do if ```yamllint.cli.run(yamllint_argv)``` fails?

**Review by jimoleary — CHANGES_REQUESTED** [review]
The ```no_format``` variable name looks a bit odd, can we change it to ```lint_format```?

This hurts my head less.

I assume you went do this path so as not to override the builtin ```format``` function.

File: docs/using.md (hunk 2 above)
Comment (jimoleary) [reply docs/using.md:453]: oh yeah, let's make that the default for now.

File: src/lamplib/src/genny/cli.py [hunk 4]
```diff
@@ -572,11 +572,14 @@ def self_test(ctx: click.Context):
 
 
 @cli.command(name="lint-yaml", help="Run pylint on all workload and phase yamls")
+@click.option(
+    "--no-format", is_flag=True, help="Do not lint the yaml formatting.", default=False, type=bool
+)
 @click.pass_context
-def lint_yaml(ctx: click.Context):
+def lint_yaml(ctx: click.Context, no_format: bool):
```
[the remark this answers is not in the document]
Comment (jimoleary) [reply src/lamplib/src/genny/cli.py:579]: lint_format defaulting to False or no_format to True would be the sensible state .

**Review by jimoleary — APPROVED** [review]
LGTM

I think we are probably on the hook for MongosyncScripts.yml, if I find out anything different after this merges ... I'll change it.

**Pull request merged**